CONN-8055 create main README (#5)
snowflakedb/connectors-native-sdk: README.md

> # Native SDK for Connectors

Reviewer: In docs we use: Snowflake Native SDK for Connectors so I would leave it as it is

> Data is fuel for data clouds. Transferring the data to Snowflake is a step where the whole data journey begins so that 
> is why, this step on the Snowflake customer's path is such an essential one.

Reviewer: I would repharase a bit: 
```suggestion
Data is fuel for data clouds. The journey of data begins with transferring it into Snowflake, because of that this is essential step for customers.
```
Or:
"Transferring the data to Snowflake is where the data journey begins, this is why this step on the Snowflake customer's path is such an essential one."

> There are many various ways of importing the data to the Snowflake environment. One of them is connectors. A connector 
> is an application that allows data flow from an external source system into Snowflake. We can also distinguish the 
> concept of a Native Connector, which in the context of Snowflake is understood as a connector application built on the 
> foundations of the [Snowflake Native App Framework][Native Apps docs]. 

Reviewer: ```suggestion
There are many various ways of importing the data to the Snowflake environment. One of them is by using connectors. A connector 
is an application that allows data flow from an external source system into Snowflake. A Native Connector is a connector built on the 
foundations of the [Snowflake Native App Framework][Native Apps docs]. 
```

**In order to accelerate and standardize the process of developing new Native Connectors, Snowflake would like to introduce
the Native SDK for Connectors which is a set of libraries, developing tools, example connectors and connector template 
projects.**

## Key components

In the diagram below, you can find key components that are listed with a brief description below the diagram.

![Repository Components Diagram](./.assets/components_diagram.png)

### Native SDK for Connectors Java

> The Java library distributed through the [Maven Central Repository][SDK in Maven Central] which consists of .sql files 

Reviewer: use the same version as in the other pr

that deliver database objects definitions that are created during the installation of the Connector in the Snowflake 
environment and a bunch of Java classes that mainly serve as procedure handlers. However, among these Java classes, 
there also are some helper/util classes that help in coping with most of the common problems that might be encountered 
during the development of the Connector project. The database objects and Java classes create a coherent whole that 
makes managing the state of the Connector much easier and lets the developers focus on the implementation of the 
specific external source ingestion logic which also is easier to do with the SDK.

Learn more from the [official documentation][Native SDK official docs].

> #### Task Reactor

Reviewer: actually why do we mention task reactor here if it's included in the jar? Using this logic we could describe every component included in SDK

This major component is built into Native SDK for Connectors Java library. It consists of .sql files and Java classes 
embedded into the library that helps in managing and executing work asynchronously, mostly the work related to data 
ingestion. This component offers stability and scalability in work executing. Although this component is dedicated to 
orchestrating the ingestion tasks, it also can be used for executing non-ingestion tasks that require an asynchronous 
way of the execution.

Learn more from the [official documentation][Task Reactor official docs]

### Native SDK for Connectors Java Test

> The separate Java library distributed through the [Maven Central Repository][SDK Test in Maven Central] which consists 
> of Java utils helping in testing the developed Connectors. These utils are custom assertions and in-memory objects 
> (mocks) for Java classes used in the Native SDK for Connectors Java library.

Reviewer: ```suggestion
A separate Java library distributed through the [Maven Central Repository][SDK Test in Maven Central] which consists 
of Java utils helping in testing the developed Connectors. These utils include custom assertions and in-memory objects 
(mocks) for Java classes used in the Native SDK for Connectors Java library.
```

### Connectors Native SDK Template

The template Gradle Java project with inbuilt Native Java SDK that allows the developer to deploy, install, and run the 
sample, mocked source connector right after downloading the template. The template is filled with some code already which 
shows how to use the Native SDK for Connectors Java according to the connector flow defined by the Native SDK for Connectors. 
Reach the [official tutorial][Template tutorial] that will guide through the whole developer flow starting
from cloning the template project, through the implementation process of key functionalities, ending with the deployed 
and running connector in the Snowflake environment!

### Example connectors

These are example projects of connectors that provide the general information on how the connectors, as Native Applications, 
should be created and deployed to the Snowflake environment. Among these connectors there is the one that is built on the 
top of the [Connectors Native SDK Template project](#connectors-native-sdk-template), to show in practice how the template 

> should be used to develop a new connector projects. There are also some basic example projects of connectors that show the 
> general concept of:

Reviewer: ```suggestion
should be used to develop a new connector project. There are also some basic examples of connectors that show the 
general concept of the native connectors:
```
